Video call documentation

m-e-l-u-h-a-n opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
issue #17588 

**Testing plan:** Checked on local dev server.


Fixes: #17588

timabbott commented:
The tip formatting is messed up:

![image](https://user-images.githubusercontent.com/2746074/112227826-7caaa400-8bed-11eb-8f79-481e11ed1fb8.png)

More interestingly, I think the structure isn't quite right.  I think the way I'd divide the content is having `start-a-call` contain to have the documentation about how to use the compose box, and the /integrations docs talk about the other configuration details (with links in both directions).  I think start-a-call probably no longer needs the tabbed widget - it can just have a bulleted list of options with each being a link to the right part of /integrations.

m-e-l-u-h-a-n commented:
@timabbott I have updated the pr and made following changes to address you comments:

* Changed `start-a-call` to just contain steps for creating and embedding video call links in messages, and a list each of whose item links to respective doc in `/integrations`.
* Modified each `/integrations` doc to contain link which links to `start-a-call` for creating video call links. It looks something like:
![image](https://user-images.githubusercontent.com/63504956/112258990-7f5cd600-8c8d-11eb-95c2-fdcd63391425.png)

and similarly for other providers.

timabbott commented on templates/zerver/help/include/disable-video-call-button.md:
```diff
@@ -0,0 +1,4 @@
+You can also disable the video call button for your organization by
+setting the provider to "None".
+
+![](/static/images/integrations/disable-video-call-button.png)
```
I would consider having this just be a section on the "start-a-call" page, since it'd be such a tiny article otherwise.

Also, https://zulip.readthedocs.io/en/latest/documentation/user.html is useful reference, but basically we try to avoid screenshots wherever possible because they are expensive to maintain.

m-e-l-u-h-a-n replied:
I actually did refer [https://zulip.readthedocs.io/en/latest/documentation/user.html](https://zulip.readthedocs.io/en/latest/documentation/user.html), but as `/integration` doc was becoming too short so I thought of this approach to also add steps about disabling the video-call button and included image so that it covers some space and looks decent, I had similar doubts about adding image but then I thought I would get appropriate instructions during review.

timabbott commented on zerver/lib/integrations.py:
```diff
@@ -504,6 +504,14 @@ def __init__(self, name: str, *args: Any, **kwargs: Any) -> None:
+        "jitsi",
+        ["communication"],
+        logo="images/integrations/logos/jitsi.svg",
+        display_name="Jitsi",
```
"Jitsi Meet", I think -- we should use the name the product uses to refer to itself.

m-e-l-u-h-a-n replied:
Made these changes.

timabbott commented on templates/zerver/integrations/jitsi.md:
```diff
@@ -0,0 +1,16 @@
+
+### Using Jitsi Meet
+
+1. **Set jitsi as Video chat provider in Organization settings**.
```
For this first step, I'd link to a /help/ article for "Set video chat provider".  Which actually is probably where the "disable video chat provider" link should go as well.

m-e-l-u-h-a-n replied:
We currently do not have a `/help` article for "Set video chat provider", there is a section `Change your video chat provider` which now points to these different providers in respective `/integrations` page. Should I add one for "Set video chat provider".

timabbott commented on templates/zerver/integrations/jitsi.md:
```diff
@@ -0,0 +1,16 @@
+
+1. **Set jitsi as Video chat provider in Organization settings**.
+
+2. [Create and embed meeting links](/help/start-a-call) in your messages.
```
I would say "You can now create Jitsi meetings using Zulip's [call button](/help/start-a-call)." for this and all the other variants.

m-e-l-u-h-a-n replied:
Yeah made these changes for `Jitsi Meet` and all other providers.

timabbott commented on templates/zerver/integrations/jitsi.md:
```diff
@@ -0,0 +1,16 @@
+also possible to use Zulip with Jitsi Meet on-premise; to configure this, just
+set `JITSI_SERVER_URL` in `/etc/zulip/settings.py`.
+
+{!disable-video-call-button.md!}
```
We definitely don't need to include this on each page.

timabbott commented:
Thanks for working on this @m-e-l-u-h-a-n! This isn't exactly what I had in mind; posted a batch of comments that should help (and it's likely useful for you to read the style guide for Help Center docs, as well).

m-e-l-u-h-a-n commented:
I have updated pr as per suggestions. Following are the changes:
* change display name for `jitsi` to `Jitsi Meet`, as suggested [here](https://github.com/zulip/zulip/pull/17726#discussion_r601192776).
* Update instruction for creating video calls as suggested [here](https://github .com/zulip/zulip/pull/17726#discussion_r601194219)

Now as rest of the suggestions were for having `Set jitsi as Video chat provider in Organization settings` point to a `/help` doc and move steps for disabling the video call feature to that `/help` doc, and I had some doubts for this which I have posted in this [comment](https://github.com/zulip/zulip/pull/17726#discussion_r601598178).

m-e-l-u-h-a-n commented:
@timabbott I have updated this pr with following changes:
* Removed the snippet I had added earlier, for disabling the video call button.

As we discussed on [chat.zulip.org](https://chat.zulip.org/#narrow/stream/19-documentation/topic/video.20call.20providers.20on.20integrations.20page/near/1143707), I have added a new section for documenting steps for changing video call provider for organization, and first step about using any provider points to that section.

Image of `/help/start-a-call` showing newely added section:
![image](https://user-images.githubusercontent.com/63504956/112759278-e53fb980-900f-11eb-8bbc-ba3cd1ae0e29.png)

Please have another look on it.

timabbott commented:
@m-e-l-u-h-a-n OK this is the right structure now.  I did a round of copy-editing to the pages, and squashed my changes into your commits.  However, I notice the choice of SVGs for the logos seems poor:

![image](https://user-images.githubusercontent.com/2746074/114645759-e79c4780-9c8e-11eb-825e-2d1e63bef45b.png)


* The Jitsi logo seems tiny
* The Zoom logo is their name, not their icon, and thus looks squished
* For BigBlueButton, we probably want this part of the logo: 
![image](https://user-images.githubusercontent.com/2746074/114645838-04387f80-9c8f-11eb-975f-84de31a64645.png)

Can you fix the SVGs (with no other changes), and post a screenshot of the result?  Otherwise, this PR is ready to merge.

m-e-l-u-h-a-n commented:
@timabbott I updated logos as suggested, here is the new result:
![image](https://user-images.githubusercontent.com/63504956/114697367-75357100-9d3b-11eb-9b6f-9c4f934c15e6.png)

May be jitsi logo would look a bit smaller but going beyond this was affecting its quality. Please have another look at it.

timabbott commented:
Yeah, this is great, merged, thanks @m-e-l-u-h-a-n!

timabbott commented:
I also will add a commit changing "video chat provider" to "Video call provider" for consistency.
